feat: Adds support for `skip_default_alerts_settings` in `mongodbatlas_organization` resource & data sources (#2933)
mongodb/terraform-provider-mongodbatlas: docs/data-sources/organization.md

* `multi_factor_auth_required` - (Optional) Flag that indicates whether to require users to set up Multi-Factor Authentication (MFA) before accessing the specified organization. To learn more, see: https://www.mongodb.com/docs/atlas/security-multi-factor-authentication/.
* `restrict_employee_access` - (Optional) Flag that indicates whether to block MongoDB Support from accessing Atlas infrastructure for any deployment in the specified organization without explicit permission. Once this setting is turned on, you can grant MongoDB Support a 24-hour bypass access to the Atlas deployment to resolve support issues. To learn more, see: https://www.mongodb.com/docs/atlas/security-restrict-support-access/.
* `gen_ai_features_enabled` - (Optional) Flag that indicates whether this organization has access to generative AI features. This setting only applies to Atlas Commercial and defaults to `true`. With this setting on, Project Owners may be able to enable or disable individual AI features at the project level. To learn more, see https://www.mongodb.com/docs/generative-ai-faq/.

> * `skip_default_alerts_settings` - (Optional) Disables automatic alert creation and defaults to `true`. When set to true, no organization level alerts will be created automatically.

Reviewer: ```suggestion
* `skip_default_alerts_settings` - (Optional) Disables automatic alert creation. When set to true, Atlas doesn't automatically create organization-level alerts. Defaults to `true`.
```

Favor active voice and present tense. Default values generally come after the 'core' description.

Reviewer 2: @maastha @lmkerbey-mdb Terraform (and other IaCs) are intentionally defaulting to true to avoid drift changes in the customer state configurations (since alerts would be created as a side-effect). Should we add a small note about to explain why we are defaulting to true?

Author: yes I think the 'why' would make sense since we diverge from API behavior which clearly states the default value to be false.

Author: Updated and added a note about we we default to true

> * `skip_default_alerts_settings` - (Optional) Flag that indicates whether to prevent Atlas from automatically creating organization-level alerts not explicitly managed through Terraform. Defaults to `true`.

Reviewer: Can you please apply this change to all the relevant locations? I will mark them.

Author: applied everywhere

> * `skip_default_alerts_settings` - (Optional) Disables automatic alert creation. When set to `true`, Atlas doesn't automatically create organization-level alerts. By default, new Atlas Organizations created with the provider have this attribute set to `true`, which prevents the creation of alerts not explicitly managed through Terraform.

Reviewer: ```suggestion
* `skip_default_alerts_settings` - (Optional) Flag that indicates whether to prevent Atlas from automatically creating organization-level alerts not explicitly managed through Terraform. Defaults to `true`.
```

I think this captures the salient information more concisely, and in a manner more aligned with the existing attribute descriptions. Apply these changes in all relevant instances.

> * `skip_default_alerts_settings` - (Optional) Flag that indicates whether to prevent Atlas from automatically creating organization-level alerts not explicitly managed through Terraform. Defaults to `true`. *Note: Organizations created with older versions of this provider may have this attribute set to `false`. In this case, users would need to manually update this attribute to `true` to prevent Atlas from automatically creating alerts.*

Author: @marcosuma added a note here, lmk if it looks good

Reviewer 2: Should we just say "once upgrading to1.31.0 or later, [...]"?
